### ISSUE TITLE
parameter attributes can be lost when creating CompoundModels from CompoundModels

### ISSUE THREAD

**Issue by chris-simpson** [astropy/astropy#8345]
When constructing CompoundModels, parameter attributes like `fixed` and `bounds` can revert to their defaults if they're created on an intermediate CompoundModel. Here's how things work (correctly) if I do everything in single steps; I'm creating a simple 2D Model with a shift in each coordinate and a rotation:

```python
>>> m1 = Shift(1)
>>> m2 = Shift(2)
>>> m3 = Rotation2D(45)
>>> m1.offset.fixed = True
>>> m12 = m1 & m2
>>> m12.offset_0
Parameter('offset_0', value=1.0, fixed=True)
>>> compound = m12 | m3
>>> compound.offset_0
Parameter('offset_0', value=1.0, fixed=True)
```

Now, here's what happens if I create the same model by constructing the 2D shift directly:
```python
>>> m12 = Shift(1) & Shift(2)
>>> m3 = Rotation2D(45)
>>> m12.offset_0.fixed = True
>>> m12.offset_0
Parameter('offset_0', value=1.0, fixed=True)
```
so we appear to be exactly where we were before, but...
```python
>>> compound = m12 | m3
>>> compound.offset_0
Parameter('offset_0', value=1.0)
```
The `fixed` attribute has gone missing, even though it's there on the model `m12` and should be propagated. `bounds` does the same.

I'm running python 3.5.4, astropy 3.1.1, numpy 1.14.2

**Comment by perrygreenfield** [comment]
I believe this is due to the current workings of CompoundModels where when being created it extracts all the information about the parameters from the constituent models, meaning any changes to parameters in subexpressions get bypassed. Fixing this is nontrivial. There will be a new version of modeling coming out that handles parameters quite differently. Basically then a change to a parameter changes both the parameter in the compound model and the constituent model simultaneously since the Parameter object is a shared instance (it isn't for current modeling).

**Comment by nden** [comment]
I think this was fixed by #8769 but feel free to reopen if there's still an issue.
